### ISSUE TITLE
Variable products prices are not shown if they were out of stock before 2.6->3.0 update and subsequently replenished

### ISSUE THREAD

**Issue by jszineap** [woocommerce/woocommerce#15566]
### EXPLANATION OF THE ISSUE

Products that were added before the update to 2.7++ and were out of stock at the time do not show their price when their stock is replenished.

The issue is solved when all existing variations of a product are deleted and subsequently readded. (That wouldn't be a problem with 10-20 products but we have over 2500 products in store...)

Screenshots (with description): 
- http://imgur.com/a/IzUjZ

Edit: I forgot, the "Enable stock management at product level" option is not checked for variable products (in both cases where it works and doesn't) 

### STEPS TO REPRODUCE THE ISSUE

- Updating from 2.6 to either 2.7, 3.0.X & 3.1 Beta-2.
- Everything is working with woocommerce 2.6. Reverting to that version also solves the problem.

### SYSTEM STATUS

<details>

`
### WordPress Environment ###

Home URL: http://metalbandtshirts.com
Site URL: http://metalbandtshirts.com
WC Version: 3.1.0
Log Directory Writable: ✔
WP Version: 4.8
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 5.6.30
PHP Post Max Size: 256 MB
PHP Time Limit: 30
PHP Max Input Vars: 2000
cURL Version: 7.53.1
OpenSSL/1.0.1e

SUHOSIN Installed: –
MySQL Version: 5.6.35
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.1.0
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (15) ###

Insert PHP: by Will Bontrager Software
LLC  – 1.3

Ninja Forms: by The WP Ninjas – 3.1.4
Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 2.2.6
Taxonomy Metadata: by mitcho (Michael Yoshitaka Erlewine)
sirzooro – 0.5

Woocommerce Advanced Bulk Edit: by George Iron – 4.2.2
WooCommerce Currency Converter: by WooCommerce – 1.6.5
WooCommerce Shipment Tracking: by WooCommerce – 1.6.8
WooCommerce: by Automattic – 3.1.0-beta.2
WooSidebars: by WooThemes – 1.4.3
WooCommerce Helper: by WooCommerce – 1.7.2 – Network enabled
Wordfence Security: by Wordfence – 6.3.10
Yoast SEO: by Team Yoast – 4.9
WP Super Cache: by Automattic – 1.4.9
YITH WooCommerce Ajax Search: by YITHEMES – 1.5.3
YITH WooCommerce Bulk Product Editing Premium: by YITHEMES – 1.2.3

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: CAD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
exclude-from-search (exclude-from-search-2)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-3 (rated-3-2)
rated-4 (rated-4)
rated-5 (rated-5)
rated-5 (rated-5-2)

### WC Pages ###

Shop base: #47 - /shop-2/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/

### Theme ###

Name: Flatsome
Version: 2.9.9
Author URL: http://www.uxthemes.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/cart/proceed-to-checkout-button.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-product_cat.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/orderby.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/success.php
flatsome/product-searchform.php
flatsome/woocommerce/single-product/add-to-cart/grouped.php
flatsome/woocommerce/single-product/add-to-cart/simple.php
flatsome/woocommerce/single-product/add-to-cart/variable.php
flatsome/woocommerce/single-product/add-to-cart/variation-add-to-cart-button.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/additional-information.php
flatsome/woocommerce/single-product/tabs/description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php

`

</details>

**Comment by addedlovely** [comment]
@jszineap had some similar issue when updating - and was related to the theme. Have you tried with a default theme or 'Storefront'.

Looks like you're theme is on version 2.9.9 - but is currently on  3.3.5 - with various releases to fix issues with WooCommerce 3.0+
https://themeforest.net/item/flatsome-multipurpose-responsive-woocommerce-theme/5484319

**Comment by mikejolley** [comment]
2 things I can think of which could be causing this;

- Transient cache
- Outofstock term on children

https://github.com/woocommerce/woocommerce/blob/c16acc6b5104acb0ed082e7df1c63dfd77598459/includes/data-stores/class-wc-product-variable-data-store-cpt.php#L153

^ Clear that transient. You can do this by going to System Status > Tools > Clear transients.

The other code which could affect this is:

https://github.com/woocommerce/woocommerce/blob/c16acc6b5104acb0ed082e7df1c63dfd77598459/includes/data-stores/class-wc-product-variable-data-store-cpt.php#L46-L76

Looking at your status report, your jumping around versions has broken some of the terms:

```
Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
exclude-from-search (exclude-from-search-2)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-3 (rated-3-2)
rated-4 (rated-4)
rated-5 (rated-5)
rated-5 (rated-5-2)
```

You have duplicates. You need to delete the duplicate terms from your database.

I also notice a bulk editor plugin in use - this could also cause errors. Look at this query:

https://github.com/woocommerce/woocommerce/blob/c16acc6b5104acb0ed082e7df1c63dfd77598459/includes/data-stores/class-wc-product-variable-data-store-cpt.php#L59-L65

If that bulk editor has broken this term, or changed the data of variations into an unrecognized format, this will fail. 

The data updater in 3.0.x which sets this term looks for meta to do so:

https://github.com/woocommerce/woocommerce/blob/c16acc6b5104acb0ed082e7df1c63dfd77598459/includes/wc-update-functions.php#L1070-L1072

I'm looking into the code to ensure variations set and unset this term correctly on save.

**Comment by jszineap** [comment]
@addedlovely The latest version of flatsome doesn't work (& 2.9.9 is kept retro-compatible with the latest WooCommerce)

@mikejolley I tried your lattest commit here https://github.com/woocommerce/woocommerce/issues/15569 by using the master branch on the store and it doesn't work.

The transient cache doesn't work either (I tried it again but forgot to mention it in my previous post).

--

I checked the database and saw the following:

For products created prior to 2.7:

The parent product is 17140 and one of the child variation is 17145:
http://imgur.com/a/xi7uj

For products recently after 2.7, the wp_term_relationships table shows:

The parent product is 24133 and the child is 24134. However, the child cannot be found in the database (not found, unlike products created prior to 2.7).
http://imgur.com/a/hCqi9

-- 

However: deleting line 17145 (the child) from the database for older product and then changing the product status to out of stock and then back in stock does solve the problem
